charts/dns-canary/templates/cronjob.yaml

```diff
@@ -39,9 +41,10 @@ spec:
                 - -c
                 - |
                   set -e
-                  TARGET="http://{{ .Values.targetService }}"
+                  TARGET="{{ .Values.targetService }}"
+                  echo "$(date) Checking DNS resolution for $TARGET"
 
-                  if curl --max-time 3 -sSf $TARGET; then
+                  if nslookup $TARGET; then
                     echo "$(date) DNS Canary check passed for $TARGET"
                   else
                     echo "$(date) DNS Canary check failed for $TARGET" >&2
```

Comment:
Let's add `[ERROR]` to the beginning of this log line. That will make it easier to set log detection rules for it.